community/assessments/projects/kubescape/self-assessment.md

```diff
@@ -0,0 +1,323 @@
+
+##### Node agent component
+
+This component has multiple purposes all bound to information available on Kubernetes nodes.
+
+* Produces SBOMs from the images avialable on the node (used by *KubeVuln*)
+* Produces information from the configurations of the Linux host of the Kubernetes node (used by *Kubescape*)
+* Creates *ApplicationProfile* using [Inspektor Gadget](https://inspektor-gadget.io) and eBPF. These profiles log the behavior of each container on the node (file access, processes launched, capabilities used, system calls done) into *ApplicationProfile* objects stored in the *Storage* component via the Kubernetes API and optionally sent to external API endpoints.
+* Creates *NetworkNeighborhood* objects using [Inspektor Gadget](https://inspektor-gadget.io) and eBPF. These profiles log the network activity of each container and they stored as objects in the *Storage* component via the Kubernetes API and optionally sent to external API endpoints.
+* Monitors container activity via eBPF and evaluates them using its own rule engine that combines static detection rules and anomaly detection to produce alerts that can be exported to AlertManager, Syslog, HTTP endpoints, STDOUT stream and other.
+
+##### Storage component
+
+This component is a Kubernetes [aggregated API extension](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) service. It stores the different objects produced by the other components and stores them on a volume as files and SQLite. It is a singleton component in the current implementation and cannot be scaled horizontaly, but it is running in 10k node clusters.
+
+##### Synchronizer component
+
+This component is an optional part of the Kubescape Operator. It enables users to replicate the Kubernetes objects in the cluster (somewhat like `rsynch`) to a remote service. It is used for collecting the Kubescape Operator objects by central services monitoring multiple clusters.
+
+#### Detailed overview
+
+For a deeper architectural overview go [here](https://github.com/kubescape/helm-charts/blob/doc/charts/kubescape-operator/README.md#in-cluster-components-overview)
+
+#### Objects
+
+* ApplicationProfile - storing workload level information about container activities (file, process, systemcall, capabilities)
+* ConfigurationScanSummary - summary object containing results of configuration scanning
+* GeneratedNetworkPolicy - object containing proposed network policies for each workload
+* KnownServer - list of IP to host conversion
+* NetworkNeighborhood - storing workload level information about network activities of each container
+* OpenVulnerabilityExchangeContainer - [VEX](https://github.com/openvex) object that was built from the vulnerability scan together with reachability analysis
+* SBOMSyftFiltered - SBOM of running containers filtered by only those that are in use
```

Comment:
hopefully this one will disappear next week

Reply:
Which one?

Reply:
filtered SBOMs